### ISSUE TITLE
MultiMC appears to choose the wrong(?) graphics drivers

### ISSUE THREAD

**Issue by Parker8283** [MultiMC/MultiMC5#1276]
Log: http://paste.ee/r/oVh9z

I had to do some trickery to use the Intel provided graphics drivers on Win8+, which have OpenGL support. Normal Minecraft launcher works just fine.

**Comment by peterix** [comment]
The real issues is: the graphics drivers are choosing the wrong graphics card to use.

This is something that needs to be fixed. How? I'm not exactly sure.

**Comment by peterix** [comment]
Let's call it a bug and actually work on that :)

**Comment by peterix** [comment]
I assume you have  dual graphics? Something like Nvidia Optimus?

Please describe what you needed to do to make it work.

**Comment by Parker8283** [comment]
wrong card? I've only got the one :P

I've got Intel integrated graphics (really old as well, but they work). I had this issue a while back, but it was because I was using the microsoft provided graphics drivers because the intel ones didn't support win8+ (this was after I updated to win10). I did some trickery to get win10 to accept the win7 intel drivers, and everything has worked just fine since then. MultiMC also worked for a while like this. I just started it today (updated to latest development release), and this started happening again.

**Comment by peterix** [comment]
Multiple issues then...

**Comment by peterix** [comment]
What happens if you rename the 'multimc.exe' to 'minecraft.exe'?

**Comment by Parker8283** [comment]
Same output

**Comment by peterix** [comment]
So, the graphics driver is completely broken. Maybe it got replaced?

**Comment by Parker8283** [comment]
The problem is, the mojang launcher still works just fine. Just checked Device Manager, and it's still using the Intel ones I installed.

**Comment by peterix** [comment]
That sounds impossible.

**Comment by Parker8283** [comment]
I'm as baffled as you are; running the vanilla launcher:
![image](https://cloud.githubusercontent.com/assets/5672271/10417327/a184528a-7005-11e5-8d9b-f50bbfad56ff.png)

Also, as I stated, MultiMC was working fine until just today, when I updated to the latest development release. Let me grab the latest stable and test again.

**Comment by Parker8283** [comment]
uhh...wat? Ok, 0.4.7 stable isn't working anymore either...I'm seriously confused.

**Comment by Parker8283** [comment]
I know that MultiMC *was* working with the intel drivers, because I used MultiMC to test the new graphics drivers and see if minecraft would launch (and it did)

**Comment by peterix** [comment]
Did the OS update the drivers while you weren't looking?

**Comment by Parker8283** [comment]
![image](https://cloud.githubusercontent.com/assets/5672271/10417389/36bb19dc-7007-11e5-81ff-8ef3842aef9a.png)

So, no.

**Comment by Parker8283** [comment]
The only thing I can think of is that either the mojang launcher is doing something weird with selecting drivers or something, or you're doing something that is somehow affecting driver selection...kinda hard to tell...

**Comment by peterix** [comment]
There is no such thing as 'driver selection'.

**Comment by peterix** [comment]
Anyway, there has to be some difference. Can you post the log from the vanilla launcher? Ideally also the json file from the version profile it is using.

**Comment by Parker8283** [comment]
Which is why I was even more confused, because I know that there IS only one driver for a device. The only think I could have thought of was it was finding the files for the other drivers on the system somewhere and using those, but that seems pretty far-fetched. At this point, I just sound like an idiot :P


[Anyways, here's a gist with that stuff.](https://gist.github.com/Parker8283/4fc915d82988266b78fa)

**Comment by peterix** [comment]
OK. I can't see anything too different in that. LWJGL version is the same. The only difference is in how it's launched.

So, I'll experiment with that. Dismantle the launcher part and get rid of some old cruft... I'll see how it goes :)

**Comment by Parker8283** [comment]
I bid you the best of luck, sir.

/me can't wait until I get my new computer.

**Comment by peterix** [comment]
Alright. If you switch to the 'dumblaunch' branch and update, you should get a much simplified launch method.

Does it work?

**Comment by Parker8283** [comment]
Nope, same result. [Here's the new log, if it helps.](http://paste.ee/r/nqbAm)

**Comment by peterix** [comment]
Right, one more time:
Close down everything, rename `MultiMC.exe` to `Minecraft.exe` and check (make sure you have file extensions enabled, so it doesn't end up being `Minecraft.exe.exe`). If that doesn't help, then I have no more ideas.

**Comment by peterix** [comment]
Actually, maybe...

Did you enable fullscreen mode in game?

**Comment by Parker8283** [comment]
Nope and nope. So...crap :/

Look, TBH, this is a really edge case thing. I'm using some really old and crappy to begin with graphics, and since you haven't gotten this before, and may never get it again, I'm not TOO worried about it. It does seem weird that the Mojang launcher can handle it though...but whatever. This is still a temporary computer, so my new one will have an ACTUAL graphics card that should have everything work just fine.

If you want to try to fix it, that'd be great, but I won't be butthurt if you just say "Yea, sorry, seriously no clue, and really not worth it."

**Comment by peterix** [comment]
Right. There are still some differences.

Mojang launcher uses its own, custom java, located here: `C:\Users\Parker & Carmen\Downloads\runtime\jre-x64\1.8.0_25`. Try to use it instead of system java.

The second difference is that there is some weird hack in the mojang launcher related to Windows 10. I assume it's simply cosmetic, because java doesn't recognize the Windows version properly. But you could try adding some bits to the java params:
```
-Dos.name=Windows 10 -Dos.version=10.0
```

**Comment by peterix** [comment]
And yes... it's bothering me that this doesn't work. And I'd like to know why.

**Comment by peterix** [comment]
Ok. That launch method change isn't going into 0.4.8. It didn't help.

**Comment by peterix** [comment]
Alright, this is most likely some sort of bad interaction between Minecraft, java 8u60 and Windows 10. Try downgrading java.

**Comment by peterix** [comment]
Related external issue: https://github.com/LWJGL/lwjgl/issues/119

**Comment by Parker8283** [comment]
Ok, so the Minecraft provided jre works just fine...and so does Java 7u80...so yea...

**Comment by peterix** [comment]
Still a problem in Java 8u71

**Comment by peterix** [comment]
The root cause has been found - it's actually the intel driver's fault. See the linked LWJGL issue.

I've posted about it on the intel driver bug report forum and it finally got some much needed attention.

**Comment by Parker8283** [comment]
"attention" being "sucks for using an old IG chipset, buy a new one" it seems...great...

**Comment by peterix** [comment]
Yeah.

You know, I'm going to work around all these issues. It's pretty much the next thing I'll do. It will be a horrible pile of weird things, but I can get all of this working.

Anyway, this is NOT a MultiMC bug.
MultiMC **CANNOT** choose the graphics drivers to use.
It can contort itself into funny shapes by impersonating the Mojang launcher and Java itself and convince the drivers to activate by faking a lot of things.

And that's what I'll do.

**Comment by peterix** [comment]
For the time being, do not use Windows 10, or get third party drivers that aren't from Intel or Microsoft.

Or switch to Linux - it has better intel graphics support that Windows.

**Comment by Parker8283** [comment]
I may or may not have mentioned this, but this is a temp computer that's not technically even mine; I fully intend on making Arch my main OS when I get my new permanent computer ;)

**Comment by peterix** [comment]
That's fine. It's aimed at anyone reading this. You aren't alone with this issue.

**Comment by Wasted31** [comment]
Hello there,
just a message to tell you I got the same problem, with Win10, included MC java version or JRE 8 and Intel HD Graphics 2000 chip : 
- no pb with official Minecraft launcher
- on MultiMC : no instance sucessfully launch / all crashes are due to LWJGL pixel format not accelerated error

After reading your posts / FAQ, I tried to : 
- rename MultiMC.exe in Minecraft.exe => not better
- test other versions of LWJGL using your Youtube tuto => not better
- update my driver version, which already was the last Intel one (but provided for Win7/8)

**Nothing worked since I installed JRE 7u80 !! Now instances launch !**

So thanks for the post :+1: , and hope this resume will help others !

**Comment by AverageComet250** [comment]
@peterix is this still supposed to be open still or is this just an issue everyone forget about and accidently didn't close

**Comment by kb-1000** [comment]
This is not really a MultiMC issue. However, it is still present, the Intel driver bug (or in Intel's eyes, probably feature) is still there.
I have a Java build that can resolve it without downgrading to 8u51 or Windows 7: https://ci.appveyor.com/project/kb1000/jdk8u/build/artifacts
It will probably soon be resolved by using Mojang's Java build by default.

**Comment by TehcJS** [comment]
solution: don't use intel
